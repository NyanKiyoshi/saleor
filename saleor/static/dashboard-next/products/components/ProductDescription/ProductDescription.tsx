import DeleteIcon from "material-ui-icons/Delete";
import EditIcon from "material-ui-icons/Edit";
import VisibilityIcon from "material-ui-icons/Visibility";
import Card, { CardContent } from "material-ui/Card";
import IconButton from "material-ui/IconButton";
import Typography from "material-ui/Typography";
import * as React from "react";

import PageHeader from "../../../components/PageHeader";
import Skeleton from "../../../components/Skeleton";
import i18n from "../../../i18n";

interface ProductDetailsCardProps {
  product?: {
    id: string;
    name: string;
    description: string;
    url: string;
  };
  onBack();
  onDelete();
  onEdit(id: string);
  onShow(url: string);
}

export const ProductDetailsCard: React.StatelessComponent<
  ProductDetailsCardProps
> = ({ onBack, onDelete, onEdit, onShow, product }) => (
  <Card>
    {product === undefined || product === null ? (
      <>
        <PageHeader onBack={onBack}>
          <IconButton onClick={() => {}} disabled>
            <VisibilityIcon />
          </IconButton>
          <IconButton onClick={() => {}} disabled>
            <EditIcon />
          </IconButton>
          <IconButton onClick={() => {}} disabled>
            <DeleteIcon />
          </IconButton>
        </PageHeader>
        <CardContent>
          <Typography>
            <Skeleton />
          </Typography>
        </CardContent>
      </>
    ) : (
      <>
        <PageHeader title={product.name} onBack={onBack}>
          <IconButton onClick={onShow(product.url)}>
            <VisibilityIcon />
          </IconButton>
          <IconButton onClick={onEdit(product.id)}>
            <EditIcon />
          </IconButton>
          <IconButton onClick={onDelete}>
            <DeleteIcon />
          </IconButton>
        </PageHeader>
        <CardContent>
          <Typography>
            {product.description
              ? product.description
              : i18n.t("No description")}
          </Typography>
        </CardContent>
      </>
    )}
  </Card>
);
export default ProductDetailsCard;
