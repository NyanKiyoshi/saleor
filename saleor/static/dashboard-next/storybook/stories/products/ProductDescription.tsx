import { storiesOf } from "@storybook/react";
import * as React from "react";

import ProductDescription from "../../../product/components/ProductDescription";

const product = {
  id: "1",
  name: "Our totally awesome book",
  description:
    "Aenean sit amet malesuada nibh. Proin nisi lorem, facilisis at tortor vel, convallis ornare nibh. In nec ipsum porta, varius leo eu, condimentum quam. Donec gravida euismod ipsum, at consequat orci efficitur nec. Phasellus lectus arcu, auctor eget porttitor eget, venenatis a lacus. Suspendisse quis urna rhoncus, commodo justo at, tempor nisl.",
  slug: "awesome-book"
};
const productWithoutDescription = {
  id: "2",
  name: "Our totally awesome cup",
  description: "",
  slug: "awesome-cup"
};

storiesOf("Products / ProductDescription", module)
  .add("with no description", () => (
    <ProductDescription
      product={productWithoutDescription}
      onBack={() => {}}
      onDelete={() => {}}
      onEdit={() => {}}
      onShow={() => {}}
    />
  ))
  .add("with description", () => (
    <ProductDescription
      product={product}
      onBack={() => {}}
      onDelete={() => {}}
      onEdit={() => {}}
      onShow={() => {}}
    />
  ))
  .add("when loading data", () => (
    <ProductDescription
      onBack={() => {}}
      onDelete={() => {}}
      onEdit={() => {}}
      onShow={() => {}}
    />
  ));
