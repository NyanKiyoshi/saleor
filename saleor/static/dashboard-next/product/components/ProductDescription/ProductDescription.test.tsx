import * as React from "react";
import * as renderer from "react-test-renderer";

import ProductDescription from "./";

const product = {
  id: "1",
  name: "Our totally awesome book",
  description:
    "Aenean sit amet malesuada nibh. Proin nisi lorem, facilisis at tortor vel, convallis ornare nibh. In nec ipsum porta, varius leo eu, condimentum quam. Donec gravida euismod ipsum, at consequat orci efficitur nec. Phasellus lectus arcu, auctor eget porttitor eget, venenatis a lacus. Suspendisse quis urna rhoncus, commodo justo at, tempor nisl.",
  slug: "awesome-book"
};
const productWithoutDescription = {
  id: "2",
  name: "Our totally awesome cup",
  description: "",
  slug: "awesome-cup"
};

describe("<ProductDescription />", () => {
  it("renders when loading data", () => {
    const component = renderer.create(
      <ProductDescription
        onBack={() => {}}
        onDelete={() => {}}
        onEdit={() => {}}
        onShow={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with no description", () => {
    const component = renderer.create(
      <ProductDescription
        onBack={() => {}}
        onDelete={() => {}}
        onEdit={() => {}}
        onShow={() => {}}
        product={productWithoutDescription}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with full data", () => {
    const component = renderer.create(
      <ProductDescription
        onBack={() => {}}
        onDelete={() => {}}
        onEdit={() => {}}
        onShow={() => {}}
        product={product}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
