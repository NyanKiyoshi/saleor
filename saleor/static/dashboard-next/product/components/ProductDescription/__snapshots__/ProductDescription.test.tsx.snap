// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDescription /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <button
      className="MuiButtonBase-root-40 MuiIconButton-root-34 MuiIconButton-colorInherit-35 Component-menuButton-30"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-39"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-43"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-49"
      />
    </button>
    <h2
      className="MuiTypography-root-56 MuiTypography-title-62 Component-title-33"
    >
      <span
        className="Component-skeleton-80"
        style={
          Object {
            "width": "10em",
          }
        }
      >
        ‌
      </span>
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-40 MuiButtonBase-disabled-41 MuiIconButton-root-34 MuiIconButton-disabled-38"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-40 MuiButtonBase-disabled-41 MuiIconButton-root-34 MuiIconButton-disabled-38"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-40 MuiButtonBase-disabled-41 MuiIconButton-root-34 MuiIconButton-disabled-38"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root-81"
  >
    <p
      className="MuiTypography-root-56 MuiTypography-body1-65"
    >
      <span
        className="Component-skeleton-80"
        style={undefined}
      >
        ‌
      </span>
    </p>
  </div>
</div>
`;

exports[`<ProductDescription /> renders with full data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <button
      className="MuiButtonBase-root-40 MuiIconButton-root-34 MuiIconButton-colorInherit-35 Component-menuButton-30"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-39"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-43"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-49"
      />
    </button>
    <h2
      className="MuiTypography-root-56 MuiTypography-title-62 Component-title-33"
    >
      Our totally awesome book
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-40 MuiIconButton-root-34"
        disabled={false}
        onBlur={[Function]}
        onClick={undefined}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
      <button
        className="MuiButtonBase-root-40 MuiIconButton-root-34"
        disabled={false}
        onBlur={[Function]}
        onClick={undefined}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
      <button
        className="MuiButtonBase-root-40 MuiIconButton-root-34"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root-81"
  >
    <p
      className="MuiTypography-root-56 MuiTypography-body1-65"
    >
      Aenean sit amet malesuada nibh. Proin nisi lorem, facilisis at tortor vel, convallis ornare nibh. In nec ipsum porta, varius leo eu, condimentum quam. Donec gravida euismod ipsum, at consequat orci efficitur nec. Phasellus lectus arcu, auctor eget porttitor eget, venenatis a lacus. Suspendisse quis urna rhoncus, commodo justo at, tempor nisl.
    </p>
  </div>
</div>
`;

exports[`<ProductDescription /> renders with no description 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <button
      className="MuiButtonBase-root-40 MuiIconButton-root-34 MuiIconButton-colorInherit-35 Component-menuButton-30"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-39"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-43"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-49"
      />
    </button>
    <h2
      className="MuiTypography-root-56 MuiTypography-title-62 Component-title-33"
    >
      Our totally awesome cup
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-40 MuiIconButton-root-34"
        disabled={false}
        onBlur={[Function]}
        onClick={undefined}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
      <button
        className="MuiButtonBase-root-40 MuiIconButton-root-34"
        disabled={false}
        onBlur={[Function]}
        onClick={undefined}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
      <button
        className="MuiButtonBase-root-40 MuiIconButton-root-34"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-39"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-43"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root-81"
  >
    <p
      className="MuiTypography-root-56 MuiTypography-body1-65"
    >
      No description
    </p>
  </div>
</div>
`;
